File: apps/frontend/lib/registry.ts
/**
 * Pre-registration registry types (goal-mcp-loop iter-30, J-18 / backlog B-901).
 *
 * Mirrors `lib/evidence.ts`'s types-plus-small-helpers pattern for the SEPARATE `GET
 * /api/research/registry` payload — every hypothesis ever registered/tested, read VERBATIM (re-format
 * only; nothing recomputed).
 *
 * This module carries NO proven-language and NO evidence-status resolution: a registration's `status`
 * ("registered" / "tested" / "closed") is a descriptive PROCESS state, never a "Proven"/"Not yet proven"
 * signal — a "tested" row may have FAILED out-of-sample (every backfilled row today did). The ONLY source
 * of "Proven" stays the certified-claims ledger via `lib/evidence.ts` / `GET /api/evidence`; this file
 * never touches that path.
 */

/** One pre-registration row, read VERBATIM from `GET /api/research/registry`. `selectors` is the EXACT
 *  cohort selector-set (`kind` + the present cohort keys + `horizon` + `direction`) the gate matches an
 *  incoming Evidence Claim against — re-displayed as-is, never recomputed or reformatted into a numeric
 *  edge. */
export interface PreRegistrationRow {
  id: string;
  selectors: Record<string, unknown>;
  rationale: string;
  registered_by: string;
  registered_date: string;
  source: string;
  /** Descriptive process state (e.g. "registered" | "tested" | "closed") — NEVER proven-language. */
  status: string;
}

/** The `GET /api/research/registry` payload: every registration, in registration (append) order. */
export interface RegistryResponse {
  registrations: PreRegistrationRow[];
}

File: apps/backend/app/api/registry.py
"""GET /api/research/registry — the read-only pre-registration registry surface (goal-mcp-loop iter-30,
J-18 / backlog B-901).

Serves `app.engine.registry.load_registrations` verbatim (re-format only — no recompute): every
hypothesis ever registered/tested, the SAME file + loader the post-decompose gate (`verify_claim.py`)
cross-checks an incoming Evidence Claim against, so the registry page a human browses and the machine
check can never disagree (the Data Contract single source of truth).

No DB/session is needed (the registry comes from the append-only state file, not the snapshot DB). The
registry path is config/env-driven via the resolver (anti-goal: No magic numbers — no path literal here).
A missing/empty registry file returns 200 with an empty list, never a 500 (anti-goal: resilience to
data-shape change) — the honest state before any backfill/registration has landed.

This module carries NO proven-language: a registration's `status` is a descriptive process state
("registered" / "tested" / "closed"), never a "Proven"/"Not yet proven" signal — that continues to flow
solely from the certified-claims ledger via `app.engine.evidence` / `GET /api/evidence`.
"""
from __future__ import annotations

from fastapi import APIRouter

from app.engine.registry import load_registrations

router = APIRouter(tags=["registry"])


@router.get("/research/registry")
def get_registry() -> dict:
    """Every registered hypothesis, verbatim, in registration (append) order: `{"registrations": [...]}`.
    READ-ONLY — recomputes nothing. An absent/empty registry file ⇒ `{"registrations": []}` (200, never
    500)."""
    return {"registrations": load_registrations()}
